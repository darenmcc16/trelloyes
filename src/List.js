import React from 'react';
import './List.css';
import Card from './Card';


//add the id portion to the <Card>
//add onClickDelete
//add onClick prop for adding a card
//define the props at the bottom 
export default function List(props){
    return(
        <section className='List'>
            <header className='List-header'>
            <h2>{props.header}</h2>
            </header>
            <div className='List-cards'>
                {props.cards.map((card) =>
                <Card
                key={card.id}
                id={card.id}
                title={card.title}
                content={card.content}
                onClickDelete={props.onClickDelete}
                />
                )}
                <button onClick={() => props.onClickAdd(props.id)} type='button' className='List-add-button'>+ Add Random</button>
            </div>
        </section>

        
    )
}

List.defaultProps = {
    onClickAdd: () => {},
}

