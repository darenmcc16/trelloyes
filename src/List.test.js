import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';


it('renders without crashing', () =>{
    const div = document.createElement('div')
    ReactDOM.render(<List key='testKey' header='testHeader' cards={[{'a':{id: 'a', title: 'First card', content: 'lorem ipsum'}}]}/>, div) 
    ReactDOM.unmountComponentAtNode(div);
})

it('renders UI as expected', () =>{
    const tree = renderer
    .create(<List key='testKey' header='testHeader' cards={[{'a':{id: 'a', title: 'First card', content: 'lorem ipsum'}}]}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
})