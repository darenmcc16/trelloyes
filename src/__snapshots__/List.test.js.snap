// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      testHeader
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3 />
      <p />
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random
    </button>
  </div>
</section>
`;
